libs/designsystem/src/lib/helpers/unique-id-generator.helper.ts

```diff
@@ -0,0 +1,19 @@
+export class UniqueIdGenerator {
+  private static registry = new Map<string, UniqueIdGenerator>();
+  private id = 0;
+
+  private constructor(private scope: string) {}
+
+  static scopedTo(scope: string) {
+    let scopedGenerator = UniqueIdGenerator.registry.get(scope);
+    if (!scopedGenerator) {
+      scopedGenerator = new UniqueIdGenerator(scope);
+      UniqueIdGenerator.registry.set(scope, scopedGenerator);
+    }
+    return scopedGenerator;
+  }
+
+  next(): string {
+    return `${this.scope}-${this.id++}`;
```

Comment:
Why not just a guid generator here?

Reply:
I don't see the need for that? Also, the 0 index based numbering is 1:1 with lists, e.g. radio buttons, checkboxes, form fields etc...